src/profiler.cpp

```diff
@@ -1075,6 +1088,30 @@ Error Profiler::start(Arguments& args, bool reset) {
     return error;
 }
 
+Error Profiler::setEcid(const char* ecid) {
+    pthread_once(&local_ecid_key_once, createEcidKey);
+
+    void *value = pthread_getspecific(local_ecid_key);
+    destroyEcidKey(value);
```

Comment:
How do you know if there is always something under this pointer that has to be freed? Even in the case when this function is being called for the first time?

Reply:
From a specification of [pthread_key_create](https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_key_create.html):

> Upon key creation, the value NULL shall be associated with the new key in all active threads. Upon thread creation, the value NULL shall be associated with all defined keys in the new thread.

Since I use ```pthread_key_create``` on both get/set I thought it is safe to assume that it is nulled. The destroy function:
```c
static void destroyEcidKey(void* data) {
    if (data) {
        free(data);
    }
}
```

So here is a check if the pointer points to something.

That's for a theory, I'm using that feature for a week, and sometimes I get some garbage as ECID, so something must not working as supposed. I put Java UUID as ECID and here is a sample of what I get:
![image](https://user-images.githubusercontent.com/57556371/163843126-69ce2e76-39c2-4619-a797-68022cac8d1a.png)

Reply:
First column is an ECID.

Reply:
To overcome this I'm post filtering the results:
https://github.com/krzysztofslusarski/jvm-profiling-toolkit/blob/309f9b3779df7e0d4f82f9d87847627f24d99124/jfr-parser/src/main/java/pl/ks/jfr/parser/JfrParserImpl.java#L336

Reply:
Code is not longer in use, resolving conversation.